src/preact/base-element.js

```diff
@@ -142,34 +143,46 @@ const childIdGenerator = sequentialIdGenerator();
  * @template API_TYPE
  */
 export class PreactBaseElement extends AMP.BaseElement {
-  /**
-   * @return {boolean}
-   * @nocollapse
-   */
+  /** @override @nocollapse */
+  static V1() {
+    return true;
+  }
+
+  /** @override @nocollapse */
   static requiresShadowDom() {
     // eslint-disable-next-line local/no-static-this
     return usesShadowDom(this);
   }
 
+  /** @override @nocollapse */
+  static prerenderAllowed() {
+    // eslint-disable-next-line local/no-static-this
+    const Ctor = this;
```

Comment:
I believe `this` is going to be undefined in production, which is why we added this lint rule. We may need to adjust `prerenderAllowed` to also pass the BaseElement ctor as a param?

Reply:
Can you explain why though? In this case the `prerenderAllowed` is not collapsed or inline or optimized in any way. As the result it's called by caller in PROD as this: `ImplClass.prerenderAllowed()`. That will always have `this` correctly defined, no? I think the issues we have with `static` and `this` is when a static method is optimized as a top-level function.

Reply:
Here's what I see in binaries:

`amp-video-1.0.js`:

```
V.prerenderAllowed=function(){return!this.loadable};
```

And it's called like this in `v0.js`:

```
b.prototype.prerenderAllowed=function(){return ... this.O?this.O.prerenderAllowed(this)};
```

The `this.O` stands for `this.implClass_`.

So from what I can tell - this all looks good. No?

Reply:
I think you're right about the issue only being for optimized toplevel funcs